hearpreprocess/pipeline.py

```diff
@@ -870,6 +890,55 @@ def run(self):
         self.mark_complete()
 
 
+class FinalizeCorpus(MetadataTask):
+    """
+    Tar the final dataset.
+
+    TODO: Secret tasks should go into another directory,
+    so we don't accidentally copy them to the public bucket.
+
+    Parameters:
+            sample_rates (list(int)): The list of sampling rates in
+                which the corpus is required.
+        tasks_dir str: Directory to put the combined dataset.
+        tar_dir str: Directory to put the tar-files.
+    Requires:
+        final_combine (FinalCombine): Final combined dataset.
+    """
+
+    sample_rates = luigi.ListParameter()
+    tasks_dir = luigi.Parameter()
+    tar_dir = luigi.Parameter()
+
+    def requires(self):
+        return {
+            "combined": FinalCombine(
+                sample_rates=self.sample_rates,
+                tasks_dir=self.tasks_dir,
+                metadata_task=self.metadata_task,
+                task_config=self.task_config,
+            )
+        }
+
+    def create_tar(self, sample_rate: str):
```

Comment:
This looks like all the sample rates end up in each of tars that are created (so we just end up with four tars that are exactly the same). It's also quite slow (quite a bit slower than the command line version https://stackoverflow.com/a/45621522) -- even for the small datasets this took several minutes. Maybe add a tqdm bar?

Reply:
https://github.com/neuralaudio/hear-preprocess/issues/39